### ISSUE TITLE
System_Boundary regression

### ISSUE THREAD

**Issue by atorian** [plantuml-stdlib/C4-PlantUML#133]
Hi there!

since ~ 13.03.2021 i have an issue with rendering diagrams on mac. Before that - same thing worked.

given diagram:

```
@startuml

!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml

System_Boundary(c1, "Some System") {

}

@enduml
```

Leads to:
```
An error has occured : java.lang.IllegalArgumentException
Winter is coming
Diagram size: 363 lines / 8634 characters.
PlantUML (1.2021.01) cannot parse result from dot/GraphViz.
Please go to https://plantuml.com/graphviz-dot to check your GraphViz version.
Java Runtime: OpenJDK Runtime Environment
JVM: OpenJDK 64-Bit Server VM
Default Encoding: UTF-8
Language: en
Country: DE
PLANTUML_LIMIT_SIZE: 4096
This may be caused by :
- a bug in PlantUML
- a problem in GraphViz
You should send this diagram and this image to
plantuml@gmail.com
or
post to
https://plantuml.com/qa
to solve this issue.
You can try to turn arround this issue by simplifing your diagram.
java.lang.IllegalArgumentException
net.sourceforge.plantuml.Dimension2DDouble.<init>(Dimension2DDouble.java:49)
net.sourceforge.plantuml.FileFormat.getJavaDimension(FileFormat.java:149)
net.sourceforge.plantuml.FileFormat.access$000(FileFormat.java:62)
net.sourceforge.plantuml.FileFormat$1.calculateDimensionInternal(FileFormat.java:125)
net.sourceforge.plantuml.graphic.StringBounderRaw.calculateDimension(StringBounderRaw.java:59)
net.sourceforge.plantuml.creole.legacy.AtomText.calculateDimension(AtomText.java:118)
net.sourceforge.plantuml.creole.Sea.add(Sea.java:60)
net.sourceforge.plantuml.creole.SheetBlock1.initMap(SheetBlock1.java:117)
net.sourceforge.plantuml.creole.SheetBlock1.calculateDimension(SheetBlock1.java:159)
net.sourceforge.plantuml.creole.SheetBlock2.calculateDimension(SheetBlock2.java:90)
net.sourceforge.plantuml.graphic.USymbolRectangle$1.calculateDimension(USymbolRectangle.java:92)
net.sourceforge.plantuml.svek.image.EntityImageDescription.calculateDimension(EntityImageDescription.java:231)
net.sourceforge.plantuml.svek.SvekNode.<init>(SvekNode.java:98)
net.sourceforge.plantuml.svek.Bibliotekon.createNode(Bibliotekon.java:63)
net.sourceforge.plantuml.svek.GeneralImageBuilder.printEntity(GeneralImageBuilder.java:547)
net.sourceforge.plantuml.svek.GeneralImageBuilder.printGroups(GeneralImageBuilder.java:606)
net.sourceforge.plantuml.svek.GeneralImageBuilder.buildImage(GeneralImageBuilder.java:410)
net.sourceforge.plantuml.svek.CucaDiagramFileMakerSvek.createFileInternal(CucaDiagramFileMakerSvek.java:114)
net.sourceforge.plantuml.svek.CucaDiagramFileMakerSvek.createFile(CucaDiagramFileMakerSvek.java:79)
net.sourceforge.plantuml.cucadiagram.CucaDiagram.exportDiagramInternal(CucaDiagram.java:655)
net.sourceforge.plantuml.classdiagram.ClassDiagram.exportDiagramInternal(ClassDiagram.java:196)
net.sourceforge.plantuml.UmlDiagram.exportDiagramNow(UmlDiagram.java:176)
net.sourceforge.plantuml.AbstractPSystem.exportDiagram(AbstractPSystem.java:145)
org.plantuml.idea.adapter.rendering.DiagramFactory.outputImage(DiagramFactory.java:123)
org.plantuml.idea.adapter.rendering.PlantUmlExporter.renderAndSave(PlantUmlExporter.java:31)
org.plantuml.idea.adapter.rendering.PlantUmlRendererUtil.renderAndSave(PlantUmlRendererUtil.java:33)
org.plantuml.idea.adapter.FacadeImpl.renderAndSave(FacadeImpl.java:34)
org.plantuml.idea.action.context.ExternalOpenDiagramAction.actionPerformed(ExternalOpenDiagramAction.java:49)
com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:281)
com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.lambda$actionPerformed$0(ActionMenuItem.java:310)
com.intellij.openapi.wm.impl.FocusManagerImpl.runOnOwnContext(FocusManagerImpl.java:286)
com.intellij.openapi.wm.impl.IdeFocusManagerImpl.runOnOwnContext(IdeFocusManagerImpl.java:77)
com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.actionPerformed(ActionMenuItem.java:299)
java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
com.intellij.openapi.actionSystem.impl.ActionMenuItem.lambda$fireActionPerformed$0(ActionMenuItem.java:110)
com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:95)
com.intellij.openapi.actionSystem.impl.ActionMenuItem.fireActionPerformed(ActionMenuItem.java:110)
com.intellij.ui.plaf.beg.BegMenuItemUI.doClick(BegMenuItemUI.java:514)
com.intellij.ui.plaf.beg.BegMenuItemUI$MyMouseInputHandler.mouseReleased(BegMenuItemUI.java:544)
java.desktop/java.awt.Component.processMouseEvent(Component.java:6652)
java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3345)
java.desktop/java.awt.Component.processEvent(Component.java:6417)
java.desktop/java.awt.Container.processEvent(Container.java:2263)
java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5027)
java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
java.desktop/java.awt.Component.dispatchEvent(Component.java:4859)
java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780)
java.desktop/java.awt.Component.dispatchEvent(Component.java:4859)
java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
java.base/java.security.AccessController.doPrivileged(Native Method)
java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
java.base/java.security.AccessController.doPrivileged(Native Method)
java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:976)
com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:911)
com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:840)
com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:454)
com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:773)
com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:453)
com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:822)
com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:507)
java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

Graphviz version:
```
dot -v
dot - graphviz version 2.46.1 (20210213.1702)
libdir = "/usr/local/Cellar/graphviz/2.46.1/lib/graphviz"
Activated plugin library: libgvplugin_dot_layout.6.dylib
Using layout: dot:dot_layout
Activated plugin library: libgvplugin_core.6.dylib
Using render: dot:core
Using device: dot:dot:core
The plugin configuration file:
        /usr/local/Cellar/graphviz/2.46.1/lib/graphviz/config6
                was successfully loaded.
    render      :  cairo dot dot_json fig gd json json0 map mp pic pov ps quartz svg tk visio vml vrml xdot xdot_json
    layout      :  circo dot fdp neato nop nop1 nop2 osage patchwork sfdp twopi
    textlayout  :  textlayout
    device      :  bmp canon cgimage cmap cmapx cmapx_np dot dot_json eps exr fig gd gd2 gif gv icns ico imap imap_np ismap jp2 jpe jpeg jpg json json0 mp pct pdf pic pict plain plain-ext png pov ps ps2 psd sgi svg svgz tga tif tiff tk vdx vml vmlz vrml wbmp webp xdot xdot1.2 xdot1.4 xdot_json
    loadimage   :  (lib) bmp eps gd gd2 gif jpe jpeg jpg pdf png ps svg webp xbm
```

**Comment by Potherca** [comment]
Since the online version (see below) seems to work, and the error message states:


> PlantUML (1.2021.01) cannot parse result from dot/GraphViz.
> Please go to https://plantuml.com/graphviz-dot to check your GraphViz version.

Could this be an issue with your (local) install?
 - - - 
 
 http://www.plantuml.com/plantuml/uml/9St1gi8m48NXVPuYjwihs0R1bQjYjmf2SLtIPB21P18c4wI8xswgs_FpSVhCchW4ByIVG-EB1RamftnIYlIpUpXUobmoa8d8WDoP65Jo6ZVKPhRUpMeujhTjkbylAkZCG7MQXYgqGw0kVG_6jPOmdMD1gsdzDuUzRCOOGFvAivCl8Tv2z82sYWy0
 
![](http://www.plantuml.com/plantuml/png/9St1gi8m48NXVPuYjwihs0R1bQjYjmf2SLtIPB21P18c4wI8xswgs_FpSVhCchW4ByIVG-EB1RamftnIYlIpUpXUobmoa8d8WDoP65Jo6ZVKPhRUpMeujhTjkbylAkZCG7MQXYgqGw0kVG_6jPOmdMD1gsdzDuUzRCOOGFvAivCl8Tv2z82sYWy0)

**Comment by atorian** [comment]
Could be, of course. Just tried to reinstall it using homebrew. The issue is still there. 
Could you please suggest what else could i try to fix it? 
BTW i use it as a plugin for IntelliJ IDE.

Tnx in advance.

**Comment by cbchenoweth** [comment]
I am seeing same behavior (also using IntelliJ plugin: https://plugins.jetbrains.com/plugin/7017-plantuml-integration).

I attempted reverting:

- the `plantuml.jar` to `v1.2021.01`
- the plugin itself back to previous version (which also reverts the `plantuml.jar` version.)

And still getting same error as @atorian.

Once I locked my import of the `C4_*.puml` files to `v2.0.1` everything is working again.
```
!includeurl https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/v2.0.1/C4_Container.puml
```

Not sure exactly where the real issue lives, but this is a current "workaround".

**Comment by Potherca** [comment]
From what I can see, the given diagram is broken in PlantUML versions before `1.2020.21` but works from `1.2020.21` onward (which would be version 2.29.1 of the Jetbrains Plugin).

I have not been able to reproduce any errors with recent (v1.2021.*) versions of PlantUML. :thinking: 

It'll take some more digging around to pinpoint where the problem lies. If anyone has more error output (generated images and/or CLI output) that would of great help!

**Comment by Potherca** [comment]
It looks like `System_Boundary` was last changed in b63db7ee9b7096c35154d081cda4ff43a07177d9, which is in the range between 2.0.1 and now, so that seems like a likely candidate.

**Comment by Potherca** [comment]
Found it!

This issue is `%newline()`, which was introduced in b63db7ee9b7096c35154d081cda4ff43a07177d9. This function was added in PlantUML `v1.2020.21`, hence everything before that breaks (as those version do not know what to make of `%newline()`).

The only solutions seem to be to either use a newer version of PlantUML or an older version of C4-PlantUML.

 - - -
For references see:

- https://forum.plantuml.net/10265/new-line-in-table-built-with-variables-is-ignored
- https://forum.plantuml.net/12480/new-line-in-table-built-with-variables-broken-from-1-2020-20
- https://plantuml.com/changes
- https://plantuml.com/preprocessing

<!--
```
for sFile in ~/Downloads/plantuml.1.202*; do echo "${sFile}";echo -e '@startuml\n%newline()\n@enduml' | java -jar "${sFile}" -pipe > /dev/null; done
/home/ben/Downloads/plantuml.1.2020.15.jar
ERROR
1
Syntax Error?
Some diagram description contains errors
/home/ben/Downloads/plantuml.1.2020.16.jar
ERROR
1
Syntax Error?
Some diagram description contains errors
/home/ben/Downloads/plantuml.1.2020.17.jar
ERROR
1
Syntax Error?
Some diagram description contains errors
/home/ben/Downloads/plantuml.1.2020.18.jar
ERROR
1
Syntax Error?
Some diagram description contains errors
/home/ben/Downloads/plantuml.1.2020.19.jar
ERROR
1
Syntax Error?
Some diagram description contains errors
/home/ben/Downloads/plantuml.1.2020.20.jar
ERROR
1
Syntax Error?
Some diagram description contains errors
/home/ben/Downloads/plantuml.1.2020.21.jar
```

**Comment by kongslund** [comment]
@Potherca I am experiencing this issue in IntelliJ using AsciiDoc plugin 0.32.26. It's using PlantUML 1.2021.00 which is newer that 1.2020.21 so there is more to the story than `%newline()` support.

I did some digging and found that it relates to `StereotypeFontSize` which sometimes has the value `0`.

E.g.

```
skinparam rectangle<<boundary>> {
    Shadowing false
    StereotypeFontSize 0
    FontColor #444444
    BorderColor #444444
    BorderStyle dashed
}
```

Here's the thing though:
* I can only reproduce an error in IntelliJ using the AsciiDoc plugin.
* I cannot reproduce the error when running the official PlantUML jar on the same content with StereotypeFontSize being 0.

My approach:

Create a test diagram `sample.puml`:
```sh
cat <<EOF > sample.puml
@startuml
!include <C4/C4_Container>

System_Boundary(c1, "Some System") {
}
@enduml
EOF
```

Output the preprocessor text to `sample.preproc`:
```
java -jar plantuml.1.2021.0.jar -preproc sample.puml
```

Observe that `sample.preproc` has `StereotypeFontSize` lines with the value 0:
```sh
grep "StereotypeFontSize 0" sample.preproc
    StereotypeFontSize 0
    StereotypeFontSize 0
```

Test that the PlantUML jar can generate `sample.png` from `sample.preproc` without problems:
```
java -jar plantuml.1.2021.0.jar sample.preproc
```

Create an AsciiDoc file `sample.adoc` with the contents of `sample.preproc`:
```sh
echo "[plantuml]\n----" > sample.adoc
cat sample.preproc >> sample.adoc
echo "----" >> sample.adoc
```

Open the `sample.adoc` in IntelliJ and see that it reproduces the error.

Now regenerate `sample.adoc` with `StereotypeFontSize` set to 12 instead of 0.
```sh
echo "[plantuml]\n----" > sample.adoc
cat sample.preproc | sed 's/StereotypeFontSize 0/StereotypeFontSize 12/g' >> sample.adoc
echo "----" >> sample.adoc
```

Check in IntelliJ that the error is no longer there and that the diagrams renders successfully.

Any idea on how StereotypeFontSize ends up being 0 in the first place?

**Comment by cbchenoweth** [comment]
after some more searching / playing around, looks like this issue is related to what we are seeing as well.
https://github.com/esteinberg/plantuml4idea/issues/261

**Comment by cbchenoweth** [comment]
follow on update here, continuing down investigation path from previous comment.

- I followed the instructions for [configuring IntelliJ to run under a different JRE](https://intellij-support.jetbrains.com/hc/en-us/articles/206544879-Selecting-the-JDK-version-the-IDE-will-run-under).
- And changed IntelliJ to run with JRE from `OpenJDK 64-Bit Server VM AdoptOpenJDK (build 15.0.2+7, mixed mode, sharing)`
- **And then the plugin started correctly rendering diagrams again.** 🎉  🥳 

So I _think_ the real root of this issue is same as what was mentioned on the issue against the plugin.  and it all rolls up to some thing with: 
- currently bundled JRE in IntelliJ having a bug (causing the `NAN` that 
- plantuml java code itself being "better able to handle `NAN` better".

But I might be 🙈 something here... 😅

**Comment by kongslund** [comment]
Using the java version bundled with IntelliJ, I am able to reproduce the issue from the command line using the PlantUML jar:
```sh
"$HOME/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/203.7717.56/IntelliJ IDEA.app/Contents/jbr/Contents/Home/bin/java" -jar plantuml.1.2021.0.jar sample.preproc
java.lang.IllegalArgumentException
	at net.sourceforge.plantuml.Dimension2DDouble.<init>(Dimension2DDouble.java:49)
	at net.sourceforge.plantuml.FileFormat.getJavaDimension(FileFormat.java:160)
	at net.sourceforge.plantuml.FileFormat.access$000(FileFormat.java:63)
	at net.sourceforge.plantuml.FileFormat$2.calculateDimension(FileFormat.java:140)
	at net.sourceforge.plantuml.creole.legacy.AtomText.calculateDimension(AtomText.java:118)
        ...
```

The java version bundled is
```
openjdk version "11.0.10" 2021-01-19
OpenJDK Runtime Environment JBR-11.0.10.8-1145.96-jcef (build 11.0.10+8-b1145.96)
OpenJDK 64-Bit Server VM JBR-11.0.10.8-1145.96-jcef (build 11.0.10+8-b1145.96, mixed mode)
```

**Comment by cbchenoweth** [comment]
cool cool, nice 🕵️  work!

Not sure what the real root cause is inside the bundled JRE that is turning `0` to `NAN` (or whatever is actually happening behind the scenes causing this explosion), but I _think_ is it's fairly safe to say that is is "outside of the scope of `C4-PlantUML` extension itself" 🤷 .

Unless we think that something in this library is feeding in bad values 🤔 .  (but seems unlikely since it works just fine with other JRE runtime.)

**Comment by atorian** [comment]
tnx everyone for the help. 
I also believe it has nothing to do with PlantUML. 
I could only suggest to create a generalized issue with some sound name and description, so people like me could easily find id and have a solution.

Yeah, NaNNaNNaNNaNNaNNaNNaN Batman!

**Comment by kongslund** [comment]
Regardless of the above findings, should we set StereotypeFontSize to 12 instead of 0?

I still wonder why StereotypeFontSize is set to 0 in [two particular cases](https://github.com/plantuml-stdlib/C4-PlantUML/blob/master/C4.puml#L54-L60) and 12 for all other cases? If I set them to 12, the diagrams I've tested look as expected _and_ they render inside IntelliJ.

It may just provide a quicker fix than waiting for IntelliJ to be released with a newer runtime that may or may not result in the issue. For reference, I've tried [switching the runtime](https://www.jetbrains.com/help/idea/switching-boot-jdk.html) to `jbrsdk-11_0_10-osx-x64-b1341.18` which according to [JetBrains Runtime Release Notes](https://confluence.jetbrains.com/display/JBR/Release+notes) is the latest official version. That didn't fix the issue. The same was also the case with an even newer version `jbrsdk-11_0_10-osx-x64-b1379.1` which is not mentioned on their Release Note page. I'm wary about changing to any other runtime which is not provided by JetBrains since it may result in other issues with the IDE.

What are your thoughts on that?

PS: For those of you looking for an even faster workaround, after your `!include <C4/C4_Container>` statement, add the following lines

```
' Workaround for https://github.com/plantuml-stdlib/C4-PlantUML/issues/133
skinparam package {
    StereotypeFontSize 12
    FontStyle plain
}

skinparam rectangle<<boundary>> {
    Shadowing false
    StereotypeFontSize 12
    FontColor $BOUNDARY_COLOR
    BorderColor $BOUNDARY_COLOR
    BorderStyle dashed
}
```

**Comment by cbchenoweth** [comment]
excellent find @kongslund, and I think you might be onto some things there.

I might be missing something, but from what I can see the reason for trying to set the `StereotypeFontSize` to `0` is because **they don't want to display `<<boundary>>` stereotype for all those boxes.**  (not sure why that decision was made, compared to all the other types always showing their `<<stereotype>>` text by default.  🤔  🤷 )

So, the one thing that workaround will impact is that you will start seeing the `<<boundary>>` text on your diagrams.  **But that can be selectively hidden by specifying `hide <<boundary>> stereotype`.**  (which actually resolves the issue for the most part all on its own, even without the other "workaround" settings.)

Also, I _think_ that the `skinparam` settings will merge with existing ones, so I think we don't need to specify all those properties for the "workaround".

So I think we could shorten the recommended "workaround" to something like:
```
' include whatever stuff from C4 you need for the diagram
!includeurl https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Context.puml
!includeurl https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml
!includeurl https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Component.puml

' Workaround for https://github.com/plantuml-stdlib/C4-PlantUML/issues/133
skinparam packageStereotypeFontSize 12
skinparam rectangle<<boundary>>StereotypeFontSize 12
hide <<boundary>> stereotype
```

**Comment by cbchenoweth** [comment]
Not sure if that is enough of a push to actually change the code in this repo.  (I could see recommendation to "not hide the `<<boundary>>` stereotype by default.  But I don't know how that decision was made in the first place, and seems unlikely they would have to "reverse" it.)

But maybe the code internally could change to use `hide <<boundary>> stereotype` instead of setting the font to `0`? 🤷

**Comment by Potherca** [comment]
Wow! This issue has gather some speed since I last looked!

Rather than speed-reed through it all, I'll go through the comments in detail this weekend and post a proper response then :+1:

**Comment by kirchsth** [comment]
`hide <<boundary>> stereotype` is a good idea, but it would change the layout (text is direct below the border)
```
@startuml
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml

skinparam packageStereotypeFontSize 6
skinparam package<<boundary>>StereotypeFontColor #770000
skinparam rectangle<<boundary>>StereotypeFontSize 6
skinparam rectangle<<boundary>>StereotypeFontColor #770000
hide <<boundary>> stereotype


System_Boundary(c1, "Sample System") {
    Container(web_app, "Web Application", "C#, ASP.NET Core 2.1 MVC", "Allows users to compare multiple Twitter timelines")
}
@enduml
```
![](http://www.plantuml.com/plantuml/png/ZP5DIyD048Rl-ol6zD92crWH2r9AQz2JbKAg7iic6Tgb-x7iJWXL_EzEK4eb5-UsCy_32_ik8yb0ZT7YIjbADpN2WSZ7-omBiattYWvDsKGCbRE4bjBAcSnhQJjd4gdMgiposyccMxskdpCZ8s7WrIvdGogB8VNC2X6FodePf04lgwFSOy4WEZfvV6AqK1y8TtrgFYzTOsiPJel5NoDtsWKOp6RNF1TYm8gatUk-sW_xhzEBEwWQuHA6UAQ54CM9dsRty7iULjCn98KqNYFyt98HV0hWEV_JiCLo9xrdyXrBM7clLIL9EPlmAX-COLLiqfV7BIi1uIQTmlejxsuhhLqReUif0ZdWahnan3IQL1UvRHKH1Y1bK7DMJ4RYIopHrbpDDm00)
but we could use a small number and change the font color to the background color (I used in the sample a different color that we can see the remaining stereotype)
```
@startuml
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml

skinparam packageStereotypeFontSize 6
skinparam package<<boundary>>StereotypeFontColor #770000
skinparam rectangle<<boundary>>StereotypeFontSize 6
skinparam rectangle<<boundary>>StereotypeFontColor #770000
' hide <<boundary>> stereotype


System_Boundary(c1, "Sample System") {
    Container(web_app, "Web Application", "C#, ASP.NET Core 2.1 MVC", "Allows users to compare multiple Twitter timelines")
}
@enduml
```
![](http://www.plantuml.com/plantuml/png/ZP5FI_D04CRl-odcJGzl2srYHIn8AQr1JrOAgNeiasHeb-wVPNT2gE9tTuDIAhaujvtdz-E1dMLWzDne9Vv9Kwcc9ZWmkt2NPHxRT2_vq9HD85zPmsGuhQpEd4BJEPF0jP9bbjzCDjtgPVsKQGnCFgvsUJHG6lAfYwmGuIYDGuyQ75P7t5CHGR9ySlGOqKA-4zpsgVcyj8sfqPyMYzz6RfNrC9ZDhk9SY9ugHhDNFRLNrdV-NFSV3h8ckCGXd7aXH76AJxsx_uc7rNGCIO7QAOBlB1d1Xu0uvvyQjbJkqBb8lb49A-UKh92bDKbSvOCnh8fD-lomZOedk4wdi7xDksobb6q3T9SAm1RYcHn6H3UAPLUvRIKpUM2fISMka8p4frYIgHkjlW00)

PS.: the $tag handling must be extended that the package stereotype color is updated too

**Comment by kirchsth** [comment]
@atorian, @cbchenoweth, @kongslund: can you please test my [fix](https://github.com/plantuml-stdlib/C4-PlantUML/pull/136) 
Helmut

**Comment by cbchenoweth** [comment]
@kirchsth - pointed to your [extended branch](https://github.com/kirchsth/C4-PlantUML/tree/extended) (like you show in your PR example) in my diagrams, and everything looks to be rendering correctly.  👍 

Checked out the code changes you are suggesting, and that seems to reasonable way to address this issue (since you are wanting to preserve the "spacing" in the layout).  👍 

Let me know if you want to give an approval on the PR.  (not sure what the contributing process is for this project.  😄 )

**Comment by atorian** [comment]
@kirchsth tnx, works as expected.

**Comment by kongslund** [comment]
@kirchsth, works nicely for me

**Comment by Potherca** [comment]
**TL;DR**: I am in favor of the change proposed by @kirchsth

 - - -
 
After reading all the comments, I think that (even though the cause of the problem lies outside this lib) we should add a fix for it here if it doesn't introduce any added problems or complexity.

In theory, this might not be desirable, put in reality this lib does not live in on isolated plain... And breaking in other tools is still a bad user experience which could be avoided by a fix _here_.

Also, setting anything to `0` in order to hide it is a bit of a hack anyway.

**Comment by atorian** [comment]
isn't it time to take an action?

**Comment by kirchsth** [comment]
@atorian: I can offer only my extended branch until all changes are approved  https://github.com/kirchsth/C4-PlantUML

**Comment by Potherca** [comment]
@atorian We are still waiting for a second reviewer to approve https://github.com/plantuml-stdlib/C4-PlantUML/pull/136. As not all contributors are available at all times, please bare with us!

**Comment by Potherca** [comment]
Closed with #136
